release-notes-agent/src/main/java/com/epam/release/notes/agent/service/action/mail/TemplateNotificationServiceImpl.java

```diff
@@ -17,15 +17,64 @@
 import com.epam.release.notes.agent.entity.github.GitHubIssue;
 import com.epam.release.notes.agent.entity.jira.JiraIssue;
 import com.epam.release.notes.agent.entity.mail.EmailContent;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
+import org.thymeleaf.context.Context;
+import org.thymeleaf.spring5.SpringTemplateEngine;
 
 import java.util.List;
+import java.util.Locale;
 
 @Service
-public class TemplateNotificationServiceImpl implements TemplateNotificationService {
+public class TemplateNotificationServiceImpl implements TemplateNotificationService{
+
+    private final SpringTemplateEngine templateEngine;
+    private final String emailToAdminTemplateName;
+    private final String emailToSubscribersTemplateName;
+    private final String emailToAdminTitle;
+    private final String emailToSubscribersTitle;
+
+    public TemplateNotificationServiceImpl(
+            final SpringTemplateEngine templateEngine,
+            @Value("${release.notes.agent.name.of.admin.email.template}") final String emailToAdminTemplateName,
+            @Value("${release.notes.agent.name.of.subscribers.email.template}")
+            final String emailToSubscribersTemplateName,
+            @Value("${release.notes.agent.email.to.admin.subject}") final String emailToAdminTitle,
+            @Value("${release.notes.agent.email.to.subscribers.subject}") final String emailToSubscribersTitle) {
+        this.templateEngine = templateEngine;
+        this.emailToAdminTemplateName = emailToAdminTemplateName;
+        this.emailToSubscribersTemplateName = emailToSubscribersTemplateName;
+        this.emailToAdminTitle = emailToAdminTitle;
+        this.emailToSubscribersTitle = emailToSubscribersTitle;
+    }
 
     @Override
-    public EmailContent populate(String oldVersion, String newVersion, List<JiraIssue> jiraIssues, List<GitHubIssue> gitHubIssues) {
-        return null;
+    public EmailContent populate(final String oldVersion,
+                                 final String newVersion,
+                                 final List<JiraIssue> jiraIssues,
+                                 final List<GitHubIssue> gitHubIssues) {
+        final Context context = new Context(Locale.US);
+        context.setVariable("oldVersion", oldVersion);
+        context.setVariable("newVersion", newVersion);
+        if (validateIssuesParameters(jiraIssues, gitHubIssues)) {
```

Comment:
maybe we need to rename this method some how or add a comment about flow of this method to be more specific and clear when we send email about a changes and when we will send email to admin